### PR TITLE
feature/ec2/imds: Move the ec2imds module

### PR DESCRIPTION
**Pull request by jasdel**
Moves the `ec2imds` module from the root of the SDK to a nested module under the `feature` folder. This improvement brings the EC2 IMDS client inline with other higher level features such as `s3/manager` and `dynamodb/attributevalue`

### CODE REVIEW

File: feature/ec2/imds/api_client.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package ec2imds
```
Comment (skotambkar) [inline feature/ec2/imds/api_client.go:1]: is this change required? Kinda like `ec2imds` parity with v1 sdk.

**Review by skotambkar — APPROVED** [review]
Would want documentation update to be paired with this change.

File: feature/ec2/imds/api_client.go (hunk 1 above)
Comment (jasdel) [reply feature/ec2/imds/api_client.go:1]: good point I'll look at the doc branch and update these.

Comment (skotambkar) [reply feature/ec2/imds/api_client.go:1]: Wait, was your ^ comment for documentation update ask,  OR on my comment to not change package name from `ec2imds` to `imds`?

Comment (jasdel) [reply feature/ec2/imds/api_client.go:1]: https://github.com/aws/aws-sdk-go-v2/pull/986
